src/aind_codeocean_api/codeocean.py

```diff
@@ -0,0 +1,315 @@
+            json_data["source"]["aws"]["secret_access_key"] = secret_access_key
+
+        response = requests.post(
+            self.asset_url, json=json_data, auth=(self.token, "")
+        )
+        return response
+
+    def update_data_asset(
+        self,
+        data_asset_id: str,
+        new_name: str,
+        new_description: Optional[str] = None,
+        new_tags: Optional[List[str]] = None,
+        new_mount: Optional[str] = None,
+    ) -> requests.models.Response:
+        """
+        This will update a data asset from a PUT request to code ocean API.
+
+        Parameters
+        ---------------
+        data_asset_id : string
+            ID of the data asset
+        new_name : str
+            New name of the data asset
+        new_description : Optional[str]
+            New description of the data asset. Default None (not updated)
+        new_tags : Optional[str]
+            New tags of the data asset. Default None (not updated)
+        new_mount : str
+            New mount of the data asset. Default None (not updated)
+        Returns
+        ---------------
+        requests.models.Response
+        """
+
+        url = f"{self.asset_url}/{data_asset_id}"
+        data = {"name": new_name}
+
+        if new_description:
+            data["description"] = new_description
+
+        if new_tags:
+            data["tags"] = new_tags
+
+        if new_mount:
+            data["mount"] = new_mount
+
+        response = requests.put(url, json=data, auth=(self.token, ""))
+
+        return response
+
+    def run_capsule(
+        self,
+        capsule_id: str,
+        data_assets: List[Dict],
+        parameters: Optional[List] = None,
+    ) -> requests.models.Response:
+        """
+        This will run a capsule/pipeline using a POST request to code ocean
+        API.
+
+        Parameters
+        ---------------
+        capsule_id : string
+            ID of the capsule
+        data_assets : List[dict]
+            List of dictionaries containing the following keys: 'id' which
+            refers to the data asset id in Code Ocean and 'mount' which
+            refers to the data asset mount folder.
+        parameters : List
+            Parameters given to the capsule. Default None which means
+            the capsule runs with no parameters.
+
+            The parameters should match in order to the parameters given in the
+            capsule.
+
+            e.g.
+            'parameters': [
+                'input_folder',
+                'output_folder',
+                'bucket_name'
+            ]
+
+            where position one refers to the parameter #1 ('input_folder'),
+            parameter #2 ('output_folder'), and parameter #3 ('bucket_name')
+
+        Returns
+        ---------------
+        requests.models.Response
+        """
+
+        data = {"capsule_id": capsule_id, "data_assets": data_assets}
+
+        if parameters:
```

Comment:
looks generic enough, but checking to be sure - this suffices to handle the trigger capsule parameter to move result data or not?